File: How to generate discrete signals/code/sine_wave_example.py
import numpy as np
import matplotlib.pyplot as plt

# Sampling Frequency Fs=1 MHz
Fs = 1e6  # I can also write 1000000
Ts = 1/Fs

# Frequency of the sine wave 100 kHz
Fc = 125e3

# Normalized Frequency
F_N = Fc/Fs

# Number of samples
len = 32

# n variable
n = np.linspace(0, len-1, len);

# Discrete Time vector
t = n*Ts   # seconds
t = t*1e6  # scaling to obtain micro seconds

# Sine wave
x = np.sin(2*np.pi*F_N*n)

# Figure
plt.figure()

plt.subplot(2,1,1)
plt.plot(n, x, 's-', label='Sine Wave')
plt.title('Sine Wave')          # Title
plt.xlabel('Samples')           # x Label
plt.ylabel('Amplitude')         # y Label
plt.legend(loc='upper right')   # Legend
plt.grid()                      # Grid

plt.subplot(2,1,2)
plt.plot(t, x, 'x-', label='Sine Wave')   # Function to perform the plot
plt.xlabel('Time [$\mu$s]')               # x Label
plt.ylabel('Amplitude')                   # y Label
plt.legend()                              # Legend
plt.grid()                                # Grid

# Display the sine wave
plt.show() 

File: How to generate discrete signals/code/sine_wave_example.m
%% Clean everything
% Clear Command Window
clc

% Remove items from workspace, freeing up system memory
clear

% Close all figures
close all   

%% Math Variables

% Sampling Frequency Fs=1 MHz
Fs = 1e6;
Ts = 1/Fs;

% Frequency of the sine wave 100 kHz
Fc = 125e3;

% Normalized Frequency
F_N = Fc/Fs;

% Number of samples
len = 32;

% n variable
n = linspace(0, len-1, len);

% Sine wave
x = sin(2*pi*F_N*n);

% Discrete Time vector
t = n*Ts;   % seconds
t = t*1e6;  % scaling to obtain micro seconds

%% Figure
figure();

subplot(2,1,1)
plot(n, x, 's-')
title('Sine Wave')          % Title
xlabel('Samples')           % x Label
ylabel('Amplitude')         % y Label
legend('Sine Wave')         % Legend
grid on                     % Grid

subplot(2,1,2)
plot(t, x, 'x-')            % Function to perform the plot
xlabel('Time [\mus]')     % x Label
ylabel('Amplitude')         % y Label
legend('Sine Wave')         % Legend
grid on                     % Grid

